hs_geographic_feature_resource/receivers.py

```diff
@@ -21,18 +21,26 @@
                                                  FieldInformationValidationForm
 from hs_geographic_feature_resource.models import GeographicFeatureResource
 
-def check_fn_for_shp(filelists):
+def is_shapefiles(files):
+    # check if uploaded files are valid shapefiles (shp, shx, dbf)
+    fn_list = []
+    for f in files:
+        fn_list.append(f.name)
+    return check_fn_for_shp(fn_list)
+
+def check_fn_for_shp(files):
+    # used by is_shapefiles
     # check a list of filenames for valid shapefiles (shp, shx, dbf)
     shp, shx, dbf = False, False, False
     all_have_same_filename = False
     shp_filename, shx_filename, dbf_filename = None, None, None
     dir_count = 0 # can have 0 or 1 folder
-    if len(filelists) >= 3: # at least have 3 files: shp, shx, dbf
-        for fpath in filelists:
-            if fpath.endswith('/'):
+    if len(files) >= 3: # at least have 3 files: shp, shx, dbf
+        for f in files:
+            if str(f).endswith('/'):
```

Comment:
@zhiyuli Can you answer this question from @pkdash whether str() is needed here?